components/Fip/index.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  ButtonV2,
+  Dialog,
+  InputV2,
+  LoadingScreen,
+  ShadowBox,
+  useWallet,
+  useWalletProvider
+} from '@glif/react-components'
+import axios from 'axios'
+import { FormEvent, useEffect, useState } from 'react'
+import { Message } from '@glif/filecoin-message'
+import * as cbor from '@ipld/dag-cbor'
+import { Header, FIPData, FormState } from './Helpers'
+
+enum Vote {
+  APPROVE = 'Approve',
+  REJECT = 'Reject',
+  ABSTAIN = 'Abstain'
+}
+
+const FIP_ID = 14
+
+export const Fip = () => {
+  const [vote, setVote] = useState<Vote>(Vote.REJECT)
+  const [fipDetails, setFipDetails] = useState<FIPData>(null)
+  const [error, setError] = useState<Error | null>(null)
+  const [formState, setFormState] = useState<FormState>(FormState.FETCHING_VOTE)
+
+  const wallet = useWallet()
+  const { getProvider, walletError } = useWalletProvider()
+
+  const onSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    setFormState(FormState.SIGNING_MESSAGE)
+    try {
+      const params = Buffer.from(cbor.encode([`${FIP_ID} - ${vote}`])).toString(
+        'base64'
+      )
+      const provider = await getProvider()
+      const newMessage = new Message({
+        to: 'f01',
+        from: wallet.robust || wallet.id,
+        nonce: 0,
+        value: '0',
+        method: 1,
+        params
+      })
+
+      setFormState(FormState.SIGNING_MESSAGE)
+      const signedMessage = await provider.wallet.sign(
+        wallet.address,
+        newMessage.toLotusType()
+      )
+      setFormState(FormState.SIGNED_MESSAGE)
+      console.log(signedMessage)
+      // comment this line in with your URL and send it...
+      // await axios.post('your-url', { ...signedMessage })
```

Comment:
@navFooh note this still needs to be commented in when i get the right URL to send